app/app.go

```diff
@@ -687,13 +687,23 @@ func (app *App) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.Respo
 	return app.mm.EndBlock(ctx, req)
 }
 
+func (app *App) EnsureBlockGasMeter(ctx sdk.Context) {
+	// TrancheKey generation and LimitOrderExpirationPurge both rely on a BlockGas meter.
+	// check that it works at startup
+	cp := app.GetConsensusParams(ctx)
+	if cp == nil || cp.Block == nil || cp.Block.MaxGas <= 0 {
+		panic("BlockGas meter must be initialized. Genesis must provide value for Block.MaxGas")
```

Comment:
How are we setting MaxGas?

Reply:
I am getting less confident in the usefulness of the gas meter for tracking time elapsed. How could it be deterministic and also useful without somehow metering each golang vm instruction?